### PR TITLE
added panel and sub group timing in SA

### LINKED ISSUE #90: i2b2 query timing in SA
related to https://github.com/ldsec/glowing-bear-medco/issues/121 

- [x]  add the option in the API for survival analysis
- [x]  add this in survival query data flow (DON'T FORGET to hard set this of start event to idenpendant)
- [x]  in CLI
- [x] in test parameter and E2E files
- [x] make some hand test and check in the logs

### PR DESCRIPTION
**Pull request by nfreundl**
I2B2 query timing was not handled in API and sub group selection procedure.

This may also solve https://github.com/ldsec/glowing-bear-medco/issues/121 . @JLRaisaro don't forget to check it really corrected the bug once this PR is closed

### CODE REVIEW

**Comment by JLRaisaro** [comment]
I will! Thanks

File: connector/client/survivalanalysis/cli.go [hunk 1]
```diff
@@ -158,7 +158,12 @@ func ExecuteClientSurvival(token, parameterFileURL, username, password string, d
 	// --- convert panels
 	timer := time.Now()
 	logrus.Info("converting panels")
```
Comment (mickmis) [inline connector/client/survivalanalysis/cli.go]: Even if not from this change, the logs here need more context, e.g. "Survival analysis <ID>: ..." 

Remember that when looking at the logs in a console everything is mixed up together. This makes debugging easier.

File: connector/server/survivalanalysis/survival_query.go [hunk 2]
```diff
@@ -139,14 +139,15 @@ func (q *Query) Execute() error {
...
 					item,
 				},
-				Not: &not,
+				Not:         &not,
+				PanelTiming: models.TimingAny,
 			}
 			panels := append(definition.Panels, newPanel)
 
 			timer = time.Now()
 			logrus.Infof("I2B2 explore for subgroup %d", i)
```
Comment (mickmis) [inline connector/server/survivalanalysis/survival_query.go]: More context for logs

File: connector/server/survivalanalysis/survival_query.go [hunk 3]
```diff
@@ -139,14 +139,15 @@ func (q *Query) Execute() error {
...
 			logrus.Tracef("panels %+v", panels)
-			patientList, err := SubGroupExplore(q.QueryName, i, panels)
+			patientList, err := SubGroupExplore(q.QueryName, i, panels, definition.SubGroupTiming)
 			if err != nil {
 				logrus.Errorf("during subgroup explore procedure: %s", err.Error())
```
Comment (mickmis) [inline connector/server/survivalanalysis/survival_query.go]: you can create once the error to avoid repeating twice the error messages

**Review by mickmis — CHANGES_REQUESTED** [review]
All good, minor comments. Also a conflict to resolve.

File: connector/server/survivalanalysis/survival_query.go (hunk 2 above)
Comment (nfreundl) [reply connector/server/survivalanalysis/survival_query.go]: From dev branches, all Info(f) Debug and Trace 's have prefixed with "Survival analysis: "

**Review by mickmis — APPROVED** [review]

**Pull request merged**